valhalla/midgard/logging.h

```diff
@@ -1,10 +1,14 @@
 #ifndef VALHALLA_MIDGARD_LOGGING_H_
 #define VALHALLA_MIDGARD_LOGGING_H_
 
-#if defined(_MSC_VER) && defined(ERROR)
+#if defined(_WIN32) && defined(ERROR)
 #undef ERROR
 #endif
 
+#if defined(__MINGW32__) && !defined(NOGDI)
+#define NOGDI
```

Comment:
the logic around NOGDI is quite screwed up.. I include it in src/CMakeLists.txt, undefine it in `tz_alt.cpp` and define it here again.. Happy to hear some better suggestions.

Reply:
I defined `NOGDI` in https://github.com/valhalla/valhalla/commit/4fb5c6e58b2cf7628f3c7311cf2fa13bc0155023 to work around WinSock issues.

Currently, this workaround is localised to single source file where it is needed https://github.com/valhalla/valhalla/blob/15a61f761ad5b37953e917e4a32bc1017db55ca4/src/baldr/curler.cc#L11-L15

I think, if the global define of `NOGDI` is removed (from the `CMakeLists.txt`) then the related logic in `tz_alt.cpp` and `logging.h` can be avoided.

Reply:
> I defined NOGDI in 4fb5c6e to work around WinSock issues.

Yeah the same is true for `logging.h` (clashes with `ERROR` macro from some win header). Probably there's a better solution for this, didn't find any other though (other than redefining the valhalla macro?).. Now I don't define it globally anymore, remove the lines in `tz_alt.cpp` and left this one `logging.h`.

Reply:
Looks good